src/main/scala/io/iohk/ethereum/jsonrpc/FilterManager.scala

```diff
@@ -124,13 +132,19 @@ class FilterManager(
       }
     }
 
+    val bestBlockNumber = appStateStorage.getBestBlockNumber()
+
     val fromBlockNumber =
-      startingBlockNumber.getOrElse(resolveBlockNumber(filter.fromBlock.getOrElse(BlockParam.Latest), appStateStorage))
+      startingBlockNumber.getOrElse(resolveBlockNumber(filter.fromBlock.getOrElse(BlockParam.Latest), bestBlockNumber))
```

Comment:
At least [geth uses the latest.](https://github.com/ethereumproject/go-ethereum/blob/master/eth/filters/api.go#L317)

I think the rationale there is the filter, if not specified, should start from the current position, ie, latest